node/resolvers/catalog/index.ts

```diff
@@ -265,10 +312,78 @@ export const queries = {
     return catalog.products(args)
   },
 
+  productsByIdentifier: async (
+    _: any,
+    args: ProductsByIdentifierArgs,
+    ctx: Context
+  ) => {
+    const {
+      clients: { catalog },
+    } = ctx
+
+    let products = [] as Product[]
+    let product = [] as Product[]
+    const { field, values } = args
+
+    switch (field) {
+      case 'id':
+        const idStart = async () => {
```

Comment:
the logic for each case in this switch is a bit repetitive—it could be it's own function, no?

Reply:
the usage would be something like:

```
case 'id':
  return doStuff(values, catalog.productById)
case 'slug':
  return doStuff(values, catalog.product)
```

Reply:
Resolved in new commit